File: java/json-transform/src/main/java/ai/bizone/jsontransform/functions/TransformerFunctionWrap.java
package ai.bizone.jsontransform.functions;

import ai.bizone.jsontransform.functions.common.*;
import co.nlighten.jsontransform.functions.common.*;

import java.util.Map;
import java.util.Optional;

public class TransformerFunctionWrap extends TransformerFunction {
    public TransformerFunctionWrap() {
        super(FunctionDescription.of(
                Map.of(
                    "prefix", ArgumentType.of(ArgType.String).position(0).defaultValue(""),
                    "suffix", ArgumentType.of(ArgType.String).position(1).defaultValue("")
                )
        ));
    }
    @Override
    public Object apply(FunctionContext context) {
        var res = context.getString(null);
        if (res == null)
            return null;
        return Optional.ofNullable(context.getString("prefix")).orElse("")
                + res
                + Optional.ofNullable(context.getString("suffix")).orElse("");
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionWrap.ts
import TransformerFunction from "./common/TransformerFunction";
import { ArgType } from "./common/ArgType";
import FunctionContext from "./common/FunctionContext";

class TransformerFunctionWrap extends TransformerFunction {
  constructor() {
    super({
      argsSets: [
        [
          { name: "prefix", type: ArgType.String, defaultValue: "" },
          { name: "suffix", type: ArgType.String, defaultValue: "" },
        ],
      ],
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const res = await context.getString(null);
    if (res == null) return null;
    return ((await context.getString("prefix")) ?? "") + res + ((await context.getString("suffix")) ?? "");
  }
}

export default TransformerFunctionWrap;
